repo/remote.go

```diff
@@ -212,6 +212,29 @@ var knownPrefixes = []struct {
 	{prefix: "github.com", missing: 2},
 }
 
+// RootStatic checks the cache to see if the provided importpath matches any known roots.
+// If no matches are found, rather than going out to the network to determine the root,
+// nothing is returned.
+func (r *RemoteCache) RootStatic(importPath string) (root, name string, err error) {
+	prefix := importPath
+	for {
+		v, ok, err := r.root.get(prefix)
```

Comment:
Is the value of `ok` still reliable when `err != nil`? Should we check `err != nil` first?

Reply:
Yeah, `err` is a cached value, and `ok` indicates whether or not there was a cache hit, so it is safe to check for `ok` first.